pyqg/tests/test_xarray_output.py

```diff
@@ -4,13 +4,85 @@
 import xarray as xr
 
 def test_xarray():
-    m = pyqg.QGModel(1)
-    ds = m.to_dataset()
-    
-    assert type(ds) == xr.Dataset
-    
-    expected_vars = ['q', 'u', 'v', 'ph', 'Qy']  
-    for v in expected_vars:
-        assert v in ds
+    expected_vars = [
+        'q', 
+        'u', 
+        'v', 
+        'ufull', 
+        'vfull', 
+        'qh', 
+        'uh', 
+        'vh', 
+        'ph', 
+        'Ubg', 
+        'Qy',
+    ]
+
+    expected_diags = [
+        'EKE', 
+        'entspec', 
+        'APEflux', 
+        'KEflux', 
+        'APEgenspec', 
+        'APEgen'
+    ]
+
+    expected_attrs = [
+        'L',
+        'W',
+        'dt',
+        'filterfac',
+        'nk',
+        'nl',
+        'ntd',
+        'nx',
+        'ny',
+        'nz',
+        'rek',
+        'taveint',
+        'tavestart',
+        'tc',
+        'tmax',
+        'twrite',
+    ]
+
+    expected_coords = [
+        'time',
+        'lev',
+        'x',
+        'y',
+        'l',
+        'k',
+    ]
     
+    # Initialize QG Model
+    year = 24*60*60*360.
+    m = pyqg.QGModel(tmax=year/2, twrite=10000, tavestart=year/3)
```

Comment:
Here we are only testing the `QGModel` class. However, there are many other models affected by this change. Ideally we would use pytest [fixture](https://docs.pytest.org/en/6.2.x/fixture.html#back-to-fixtures) to parametrize the test over each different model. That way we could make sure that at least the new function does not raise any errors.

Reply:
Good idea! I've started on this, but my commit will likely fail since I'm not super familiar with pytest fixtures. Please let me know if I'm on the right track.

Reply:
> Fixture "QGModel" called directly. Fixtures are not meant to be called directly,
> but are created automatically when test functions request them as parameters.

How do you parameterize the fixtures in this case?

Reply:
Just saw this question now! I saw your latest commits, and it looks like you're still investigating this.

The trick is to use a [parametrized fixture](https://docs.pytest.org/en/6.2.x/fixture.html#parametrizing-fixtures).

First just write some standalone regular functions that set up the models, e.g. like you already have (but without the decorator

```python
def SQG():
    '''Initialize Surface Quasi-Geostrophic Model'''
    return pyqg.SQGModel(L=2.*np.pi, nx=512, tmax = 26.005, beta = 0., 
                           Nb = 1., H = 1., rek = 0., rd = None, dt = 0.005,
                           taveint=1, twrite=400, ntd=4)

def two_layer():
    # etc
```

Then write a parametrized fixture like this
```python
@fixture(params=[SQG, two_layer, ...])
def all_models(request)
    # actually call the function
    model = request.param()
    return model
```

Then use this as the input to your test
```python
def test_xarray(all_models):
```

🤯

Reply:
🙏  thank you!